bin.src/purge_topics.py

```diff
@@ -0,0 +1,120 @@
+# This file is part of ts_salobj.
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import argparse
+import asyncio
+import importlib
+
+from lsst.ts import salobj
+
+
+SLEEP_TIME = 0.5  # time to sleep after subscribe and before closing the manager
+
+
+async def main():
+    parser = argparse.ArgumentParser(
+        description="Purge SAL Command Messages for a CSC component. "
+        "This is a workaround for a SAL bug."
+        "If no commands, events or telemetry are specified "
+        "then all topics are purged")
+    parser.add_argument("component",
+                        help="Name of SAL component (e.g. ATDome).")
+    parser.add_argument("-i", "--index", default=0,
+                        help="Index of SAL component.")
+    parser.add_argument("-c", "--commands", nargs="+", default=None,
+                        help="Command topics to purge.")
+    parser.add_argument("-e", "--events", nargs="+", default=None,
+                        help="Event topics to purge.")
+    parser.add_argument("-t", "--telemetry", nargs="+", default=None,
+                        help="Telemetry topics to purge.")
+    args = parser.parse_args()
+
+    await purge_topics(component=args.component,
+                       index=int(args.index),
+                       commands=args.commands,
+                       events=args.events,
+                       telemetry=args.telemetry)
+
+
+async def purge_topics(component, index=0, commands=None, events=None, telemetry=None, sleep=SLEEP_TIME):
+    """Purge commands for a given SAL component.
+
+    Parameters
+    ----------
+    component : `str`
+        Name of SAL component, e.g. "ATDome".
+    index : `int`
+        Index of SAL component.
+    commands : `list` of `str` (optional)
+        Names of comamnds to purge; if None then purge all commands
+    sleep : `float`
+        Amount of time to sleep after purging each command,
+        before closing the SAL object (sec).
+    """
+    lib = importlib.import_module(f"SALPY_{component}")
+    if commands is None and events is None and telemetry is None:
+        salinfo = salobj.SalInfo(lib, index)
```

Comment:
I think `salinfo` has to be define before the `if` statement, otherwise the script will fail if any one of the commands, events or telemetry is passed.